### PR TITLE
[MODULAR] Restores the 'feign impairment' verb

### PR DESCRIPTION
**Pull request by desertmote**
## About The Pull Request

I also included the blush you gain when drunk.

## How This Contributes To The Skyrat Roleplay Experience

We used to have this until TG updated how status effects are handled.

## Changelog

:cl:
fix: Restores the 'feign impairment' verb
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -103,4 +103,41 @@
...
+		if("jittering")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/jitter, only_if_higher = TRUE)
+		if("stupidness")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/speech/stutter/derpspeech, only_if_higher = TRUE)
+
+	if(duration)
+		addtimer(CALLBACK(src, .proc/acting_expiry, impairment), duration SECONDS)
+		to_chat(src, "You are now feigning [impairment].")
+
+/mob/living/carbon/human/proc/acting_expiry(var/impairment)
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]:
```suggestion
/mob/living/carbon/human/proc/acting_expiry(impairment)
```

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -103,4 +103,39 @@
...
+/mob/living/carbon/human/verb/acting()
+	set category = "IC"
+	set name = "Feign Impairment"
+	set desc = "Pretend to be impaired for a defined duration."
+
+	if(stat != CONSCIOUS)
+		to_chat(usr, span_warning("You can't do this right now..."))
+		return
+
+	var/list/choices = list("drunkenness", "stuttering", "jittering", "stupidness")
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]:
```suggestion
	var/static/list/choices = list("drunkenness", "stuttering", "jittering", "stupidness")
```

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -103,4 +103,39 @@
...
+	switch(impairment)
+		if("drunkenness")
+			SEND_SIGNAL(usr, COMSIG_ADD_MOOD_EVENT, "drunk", /datum/mood_event/drunk)
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/speech/slurring/drunk, only_if_higher = TRUE)
+		if("stuttering")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/speech/stutter, only_if_higher = TRUE)
+		if("jittering")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/jitter, only_if_higher = TRUE)
+		if("stupidness")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/speech/stutter/derpspeech, only_if_higher = TRUE)
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: not hugely a fan of this
```
	message = replacetext(message, " am ", " ")
	message = replacetext(message, " is ", " ")
	message = replacetext(message, " are ", " ")
	message = replacetext(message, "you", "u")
	message = replacetext(message, "help", "halp")
	message = replacetext(message, "grief", "grife")
	message = replacetext(message, "space", "spess")
	message = replacetext(message, "carp", "crap")
	message = replacetext(message, "reason", "raisin")
```

Comment (desertmote) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: Its one of the mild traumas you can get when you hit your head, called /datum/brain_trauma/mild/dumbness. this part of the datum is most relevant:
```
if(prob(capitalize_prob))
		var/exclamation = pick("!", "!!", "!!!")
		message = uppertext(message)
		message += "[apply_speech(exclamation, exclamation)]"
```
I understand that some of those mentioned replacements are a bit cringe though...

Comment (tf-4) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: Yeahhhh. If you want to adjust it to be a bit less uh. `"you" -> "u"`. Then that would be good. The exclamation stuff is fine enough, but it kind of generally just feels like something you can do ICly? Whereas stuff like drunkness is more complex and as a result harder, and obviously jittering is not realistically possible.

Comment (desertmote) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: Actually, let me see if I can find the correct datum for the brain trauma which scrambles word placement in a sentence, and then use that for stupidness instead.

Comment (tf-4) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: that would be preferable yeah

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -103,4 +103,50 @@
...
+		if("jittering")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/jitter, only_if_higher = TRUE)
+		if("stupidness")
+			set_timed_status_effect(duration SECONDS, /datum/status_effect/speech/unintelligible, only_if_higher = TRUE)
+
+	if(duration)
+		addtimer(CALLBACK(src, .proc/acting_expiry, impairment), duration SECONDS)
+		to_chat(src, "You are now feigning [impairment].")
+
+/mob/living/carbon/human/proc/acting_expiry(impairment)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm:138]: Please document this

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -103,4 +103,50 @@
...
+	if(impairment)
+		if(impairment == "drunkenness")
+			SEND_SIGNAL(usr, COMSIG_CLEAR_MOOD_EVENT, "drunk")
+
+		to_chat(src, "You are no longer feigning [impairment].")
+
+/datum/status_effect/speech/unintelligible/handle_message(datum/source, list/message_args)
+	var/message = html_decode(message_args[1])
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: That `1` should be a define to represent what it actually stands for here.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -103,4 +103,50 @@
...
+	// This is simulating the TRAIT_UNINTELLIGIBLE_SPEECH without actually giving the trait
+	message = unintelligize(message)
+
+	message_args[1] = message
+
+	var/mob/living/living_source = source
+	if(!isliving(source) || living_source.has_status_effect(/datum/status_effect/speech/unintelligible))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: What's the point of this check at the end of the proc, exactly?

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm [hunk 7]
```diff
@@ -103,4 +103,50 @@
...
+	if(stat != CONSCIOUS)
+		to_chat(usr, span_warning("You can't do this right now..."))
+		return
+
+	var/static/list/choices = list("drunkenness", "stuttering", "jittering", "stupidness")
+	var/impairment = tgui_input_list(src, "Select an impairment to perform:", "Impairments", choices)
+	if(!impairment)
+		return
 
+	var/duration = tgui_input_number(src, "How long would you like to feign [impairment] for?", "Duration in seconds", 25, 36000)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human.dm]: Those two numerical values should be defines instead.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Comment by desertmote** [comment]
Oh boy I love copying years old code and getting reviews on it 😄ill check it out soon
edit: I think I'll end up removing the newest option I added, because i have a hunch TG will eventually refactor 'unintelligize' (aka the old code i just copied)

**Review by jjpark-kb — CHANGES_REQUESTED** [review]
there should be a way to cancel fake impairments... or to make sure that this verb isnt a way to override actual impairments.

**Comment by desertmote** [comment]
> there should be a way to cancel fake impairments... or to make sure that this verb isnt a way to override actual impairments.

1st part no i am on hiatus
2nd part yes its already like that

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**